Rubberduck.Parsing/VBA/RubberduckParser.cs

```diff
@@ -72,10 +70,26 @@ private void ReparseRequested(object sender, ParseRequestEventArgs e)
                 {
                     ParseAsync(e.Component, CancellationToken.None).Wait();
 
-                    if (_state.Status != ParserState.Error)
+                    if (_resolverTokenSource.IsCancellationRequested || _central.IsCancellationRequested)
+                    {
+                        return;
+                    }
+
+                    if (_state.Status == ParserState.Error) { return; }
+
+                    var qualifiedName = new QualifiedModuleName(e.Component);
+                    Logger.Debug("Module '{0}' {1}", qualifiedName.ComponentName,
+                        _state.IsNewOrModified(qualifiedName) ? "was modified" : "was NOT modified");
+
+                    _state.SetModuleState(e.Component, ParserState.Resolving);
+                    ResolveDeclarations(qualifiedName.Component,
+                        _state.ParseTrees.Find(s => s.Key == qualifiedName).Value);
+
+                    _state.SetStatusAndFireStateChanged(ParserState.ResolvedDeclarations);
+
+                    if (_state.Status < ParserState.Error)
```

Comment:
Wouldn't changing this to `if (_state.Status == ParserState.Parsed)` fix a lot of issues?